File: app/src/main/java/br/svcdev/githubclient/presenter/Presenter.java
package br.svcdev.githubclient.presenter;

import br.svcdev.githubclient.model.Model;
import br.svcdev.githubclient.view.MainView;

public class Presenter {

	private MainView view;
	private Model model = new Model();

	public Presenter(MainView view) {
		this.view = view;
	}

	public void onCounterClick(int index) {
		view.setButtonText(index, String.valueOf(model.incrementValue(index)));
	}

}

File: app/src/main/java/br/svcdev/githubclient/presenter/Presenter.kt
package br.svcdev.githubclient.presenter

import br.svcdev.githubclient.model.ButtonIndexes
import br.svcdev.githubclient.model.Model
import br.svcdev.githubclient.view.MainView

class Presenter(private val view: MainView) {
    private val model = Model()

    fun onButtonOneClick() {
        view.setButtonOneText(model.incrementValue(ButtonIndexes.ONE).toString())
    }

    fun onButtonTwoClick() {
        view.setButtonTwoText(model.incrementValue(ButtonIndexes.TWO).toString())
    }

    fun onButtonThreeClick() {
        view.setButtonThreeText(model.incrementValue(ButtonIndexes.THREE).toString())
    }
}